com.microsoft.mrtk.uxcore/Dialog/DialogSpawner.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Serialization;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// Spawns dialogs with the requested parameters and manages
+    /// the lifecycle of the resulting dialog component.
+    /// </summary>
+    [ExecuteAlways]
+    [AddComponentMenu("MRTK/UX/Dialog Spawner")]
+    public class DialogSpawner : MonoBehaviour
+    {
+        /// <summary>
+        /// Specifies the <see cref="DialogSpawner"/>'s behavior
+        /// when opening a dialog while one is already active.
+        /// </summary>
+        public enum Policy
```

Comment:
Are we not allowing multiple dialogs to be spawned at the same time, or does each one need a separate spawner to manage it?

Reply:
It enforces only one dialog globally. I experimented with multiple dialogs, but it really isn't meaningful to the user to have multiple dialogs when they're not actually modally stacked (as we don't have the tech for that yet) and we also don't have the tech to enforce relative ordering and have them not collide/clip each other/look horrible.